### ISSUE TITLE
Error with bowtie/bismark on Mox

### ISSUE THREAD

**Issue by yaaminiv** [RobertsLab/resources#742]
I tried running [this script](https://github.com/RobertsLab/project-gigas-oa-meth/blob/master/scripts/2019-09-03-Bismark.sh) on Mox but got the following `bowtie` error. 

<img width="1042" alt="Screen Shot 2019-09-09 at 1 08 44 PM" src="https://user-images.githubusercontent.com/22335838/64562927-f2e4ff80-d302-11e9-8b3b-cc17630f305b.png">

My script uses `-path_to_bowtie2` not `path_to_bowtie`, and it doesn't have that many lines, so I'm unsure why I get this error.

Based on the last line in the screenshot, I'm not sure if it's recognizing my wildcard for the trimmed files either...any reason why that's happening? It works in [this notebook]():

<img width="660" alt="Screen Shot 2019-09-09 at 1 11 54 PM" src="https://user-images.githubusercontent.com/22335838/64563121-62f38580-d303-11e9-889e-c1e86a0bec53.png">

**Comment by sr320** [comment]
Here is code that works...
https://genefish.wordpress.com/2019/08/28/codecat-20190814_bmrkcalig-sh-bin-bash-job/

Can't explain but try `-path_to_bowtie`

**Comment by yaaminiv** [comment]
My `bowtie` error was resolved when I removed all path shortcuts and went from `bismark 0.21.0` to `bismark 0.19.0`. Still getting an issue with it recognizing wildcards in `find`:

<img width="1043" alt="Screen Shot 2019-09-09 at 1 53 25 PM" src="https://user-images.githubusercontent.com/22335838/64565699-30e52200-d309-11e9-9a78-689171015f7d.png">

I got a "no job control" error (see screenshot). If I don't include `%%bash` I don't get that error. How do I get it to recognize my wildcard for `find | xargs`? @shellytrigg have you run into this issue before? I haven't so I'm confused!

**Comment by yaaminiv** [comment]
If I run the find/xargs commands outside of the script, it works just fine:

<img width="993" alt="Screen Shot 2019-09-09 at 1 57 44 PM" src="https://user-images.githubusercontent.com/22335838/64566038-e44e1680-d309-11e9-8e44-a23a1573e845.png">

Not sure why it's not working in the script

**Comment by sr320** [comment]
`%%bash` is not needed....

here is example of mine. https://sr320.github.io/Geoduck-alignments-v74/

**Comment by yaaminiv** [comment]
I changed `find ${trimmed_files}YRV*_R1_001.fastq.gz` to `find ${trimmed_files}*_R1_001.fastq.gz` and got the script to process the wildcards correctly. Not sure why that worked, but it does!

**Comment by yaaminiv** [comment]
I ran [this script](https://github.com/RobertsLab/project-gigas-oa-meth/blob/master/scripts/2019-09-03-Bismark.sh) and got one of my samples "YRVA" to align, deduplicate, sort, and index. However, I could not get "YRVL" to align. I tried running just the "YRVL" alignment with [this script](https://github.com/RobertsLab/project-gigas-oa-meth/blob/master/scripts/2019-09-11-Bismark-YRVL-Align.sh) and got this error:

```
Use of uninitialized value $path_to_bowtie in concatenation (.) or string at /gscratch/srlab/programs/Bismark-0.21.0/bismark line 6893.
Failed to execute Bowtie 2 porperly (return code of 'bowtie2 --version' was 32512). Please install Bowtie 2 or HISAT2 first and make sure it is in the PATH, or specify the path to the Bowtie 2 with --path_to_bowtie2 /path/to/bowtie2, or --path_to_hisat2 /path/to/hisat2
```

I changed `path_to_bowtie` to `path_to_bowtie2` and got the same error. I even tried creating a new working directory and running my original script fresh, and I still run into this error. I know I got the script to run beforehand using the latest version of `bismark` installed on Mox (job 1300439)...why is it failing now?

**Comment by sr320** [comment]
What is the path to the script on mox?

thanks
Steven


Steven B. Roberts, Associate Professor
Associate Director - Graduate Program Coordinator
School of Aquatic and Fishery Sciences
University of Washington
Fisheries Teaching and Research (FTR) Building - Office 232
1140 NE Boat Street - Seattle, WA 98105
robertslab.info - sr320@uw.edu - @sr320
vm:206.866.5141 - cell:360.362.3626

**Comment by yaaminiv** [comment]
all scripts are under `/gscratch/srlab/yaaminiv/`

**Comment by sr320** [comment]
This code works
https://d.pr/n/qaUQJA


likely as you did not have

source /gscratch/srlab/programs/scripts/paths.sh

**Comment by yaaminiv** [comment]
I will run that from my account then. Thanks!

**Comment by kubu4** [comment]
Just to add to this. This was a known problem (due to a small typo in the code) in `bismark 0.21.0`. It was/is resolved in `Bismark-0.21.0_dev`, which is also on Mox. The error is mentioned in the changelog.md file in the `Bismark-0.21.0_dev` directory.
